src/router.ts

```diff
@@ -97,15 +99,24 @@ export class OneSchemaRouter<
 
   implement<Route extends keyof Schema & string>(
     route: Route,
-    implementation: EndpointImplementation<
-      Route,
-      z.output<Schema[Route]['request']>,
-      z.infer<Schema[Route]['response']>,
-      R
-    >,
-  ) {
+    ...middlewares: [
+      ...OneSchemaRouterMiddleware<
+        ContextOfEndpoint<Route, z.output<Schema[Route]['request']>, R>
+      >[],
+      EndpointImplementation<
+        Route,
+        z.output<Schema[Route]['request']>,
+        z.infer<Schema[Route]['response']>,
+        R
+      >,
+    ]
+  ): this {
     const endpoint = this.schema[route];
 
+    const mws = middlewares.slice(0, -1) as any[];
+
+    const implementation = middlewares.at(-1) as any;
```

Comment:
What was the reason for having to cast as any here?

Reply:
`middlewares` doesn't narrow enough -- it's typed as roughly `Array<Middleware | Implementation>`